### DISCUSSION TITLE
Use either rms slope or rms height with fourier_synthesis()?

### DISCUSSION THREAD

**Discussion by sitangshugk95** [ContactEngineering/SurfaceTopography#362] [Q&A]
If I use [`fourier_synthesis()`](https://github.com/ContactEngineering/SurfaceTopography/blob/1ff35591a64acbde507f896a6a70460cb1f87afb/SurfaceTopography/Generation.py#L158) to generate a random surface, and then validate whether the generated surface really has the rms height/slope that I passed as I/P arguments using `rms_height_from_area()`, it looks like it respects the conditions if I define either the slope or the roughness, but not both. 

If I define both simultaneously, it seems to only respect the one of them (the height or the slope) - is this expected, based on the algorithm that is implemented to generate the surface?

**Comment by pastewka** [comment]
It should give an error if you specify both. This sounds like a bug.

**Reply by sitangshugk95** [reply]
It did not give an error (at least on the version that I have locally, which shouldn't be very old but I need to check). Regardless, the bottom line is that I need to define either the slope or the height (but not both) - is that right?

**Comment by pastewka** [comment]
Correct
